app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -35,27 +49,49 @@
     private DrawerLayout mDrawerLayout;
     private Toolbar mToolbar;
     private NavigationView navigationView;
+    private ProgressBar mProgress;
+    private FrameLayout frameLayout;
+    private int counter;
+    private int eventsDone;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
-
-        DataDownload download = new DataDownload();
-        download.downloadVersions();
-
+        eventsDone = 0;
         setUpToolbar();
         setUpNavDrawer();
+        frameLayout = (FrameLayout) findViewById(R.id.layout_main);
         navigationView = (NavigationView) findViewById(R.id.nav_view);
+        mProgress = (ProgressBar) findViewById(R.id.progress);
+        mProgress.setVisibility(View.VISIBLE);
+        mProgress.setIndeterminate(true);
+        Bus bus = OpenEventApp.getEventBus();
+        bus.register(this);
+        DataDownload download = new DataDownload();
+        download.downloadVersions();
 
         this.findViewById(android.R.id.content).setBackgroundColor(Color.LTGRAY);
+
         FragmentManager fragmentManager = getSupportFragmentManager();
         fragmentManager.beginTransaction()
                 .replace(R.id.content_frame, new TracksFragment()).commit();
 
     }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Bus bus = OpenEventApp.getEventBus();
+//        bus.register(this);
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        Bus bus = OpenEventApp.getEventBus();
+//        bus.unregister(this);
```

Comment:
If you try to register twice with the bus, it will obviously crash. And if
you unregister without having registered it will also crash. This is about
sane programming. You need to properly implement register and unregister in
the correct life cycle events. If you do register in onResume, then u
should unregister in onPause, as they are non overlapping lifecycle events.
If you are doing it in onCreate then you should unregister it in onDestroy.
It is important that you stick to the lifecycle of the object involved (in
this case MainActivity) otherwise there will be memory leaks.

On Mon, Aug 10, 2015 at 3:02 AM, Manan Wason <notifications@github.com>
wrote:

> In app/src/main/java/org/fossasia/openevent/activities/MainActivity.java
> <https://github.com/fossasia/open-event-android/pull/69#discussion_r36592333>
> :
>
> >          FragmentManager fragmentManager = getSupportFragmentManager();
> >          fragmentManager.beginTransaction()
> >                  .replace(R.id.content_frame, new TracksFragment()).commit();
> >
> >      }
> >
> > +    @Override
> > +    protected void onResume() {
> > +        super.onResume();
> > +        Bus bus = OpenEventApp.getEventBus();
> > +//        bus.register(this);
> > +    }
> > +
> > +    @Override
> > +    protected void onPause() {
> > +        super.onPause();
> > +        Bus bus = OpenEventApp.getEventBus();
> > +//        bus.unregister(this);
>
> The app was crashing on doing so. I have to figure out why it was crashing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fossasia/open-event-android/pull/69/files#r36592333>.
>



-- 
Mohit Singh KANWAL (Mr)

[image: Facebook] <http://facebook.com/mohitkanwal>  [image: Twitter]
<http://twitter.com/mohitkanwal>